From 54e8c7a60af87c96418c6bde75e4e9e7ff67b0ef Mon Sep 17 00:00:00 2001
From: Chetan Gurjar <chetan.gurjar@mediatek.com>
Date: Wed, 24 Jul 2019 19:35:33 +0800
Subject: [PATCH] Correction in logic of roundend size calculation of SD card

API FileUtils.roundStorageSize() rounded up the Storage size incorrectly
and caused the storage total size to be displayed twice the actual capacity
in storage Settings application.
Modifications done to appropriately roundup the storage size in the
multiple of power of 2 by making the loop iterations to be controlled by
1024 instead of 1000.

Test: Verified with the particular SD card with which bug reported.
Change-Id: I8f92248a457d6ac507d092df46e1cb0daaa55134
(cherry picked from commit 8241472871287b56fee3831fec61a4b890494c3f)
---
 core/java/android/os/FileUtils.java | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/core/java/android/os/FileUtils.java b/core/java/android/os/FileUtils.java
index edfcb3d6f12a..0621ebc9d942 100644
--- a/core/java/android/os/FileUtils.java
+++ b/core/java/android/os/FileUtils.java
@@ -1299,11 +1299,13 @@ public final class FileUtils {
     public static long roundStorageSize(long size) {
         long val = 1;
         long pow = 1;
-        while ((val * pow) < size) {
+        long pow1024 = 1;
+        while ((val * pow1024) < size) {
             val <<= 1;
             if (val > 512) {
                 val = 1;
                 pow *= 1000;
+                pow1024 *= 1024;
             }
         }
         return val * pow;
-- 
2.33.0

